src/multiplex/client.rs

```diff
@@ -488,15 +469,12 @@ where
                         tracing::trace!("request sent");
                         drop(guard);
 
+                        this.in_flight.fetch_add(1, atomic::Ordering::AcqRel);
```

Comment:
Should we move this into the `Pending` constructor so that `Pending` owns both the increment and the decrement?

Reply:
Yeah I think that is a good idea.